bash_kernel/kernel.py

```diff
@@ -129,6 +135,53 @@ def process_output(self, output):
 
     def do_execute(self, code, silent, store_history=True,
                    user_expressions=None, allow_stdin=False):
+
+        if code.startswith('#DC_GRADE'):
```

Comment:
Can't you introspect the metadata of the cell in order to know if it's a test cell?

Reply:
I'm not sure. It seems like it should be possible, but I don't know how to simply introspect the full cell data :/

Reply:
If we slot it in later it will be backwards compatible though ;)